### ISSUE TITLE
Choose/Change the color at "Mark" search!

### ISSUE THREAD

**Issue by Hosen1989** [notepad-plus-plus/notepad-plus-plus#4535]
### The Request

It will be more helpful if we would be able to choose different color when we
try marking a specific text rather than only red color (diffrent color at the same time).

Yes, sure we can change the Marking color by going to

> (menu) -> Style Configurator…

but its not what I’m trying to ask.
![Style Configurator](https://i.imgur.com/insDDLL.png)

**What I try to say is**: choosing any color in Marking dialog (without going to **stylers.xml**),
And have different color highlighted/Marked in our text at the same time.
![Marking](https://i.imgur.com/mpNC0kz.png)

**Comment by dinkumoil** [comment]
Do you mean this?

![grafik](https://user-images.githubusercontent.com/24699812/40588674-96d38876-61e1-11e8-9864-5dd9ac3abef0.png)

Also available via main menu:

![grafik](https://user-images.githubusercontent.com/24699812/40588668-6683a804-61e1-11e8-8933-e8273c7a76bf.png)

**Comment by sasumner** [comment]
@dinkumoil I think that @Hosen1989 wants more than one color available to indicate the results of a regular expression search.  The 1st - 5th styles you pointed out provide multiple colors, but the text that is colored would be constant with that feature.  The true "Mark..." feature gives the variable text functionality but only with red (default) coloring.  Imagine that feature, but opened up to multiple colors.  It would have been clarifying if @Hosen1989  had proposed how the coloring should be specified in a hacked-up Find/Mark window.

**Comment by dinkumoil** [comment]
@sasumner 

You are right, I presumably missed the real intention of @Hosen1989 's feature request.

**Comment by Hosen1989** [comment]
@dinkumoil This option is some how static and and have some limitation like (I heavily using it):

- You can not use regex in Marking by this way.
- You have only 5 color style.

But in the other hand, you have different search mode in **"Mark"** dialog, And the best one is that you can mark and highlight any **pattern using regex**.

@sasumner Yes, you are correct and sorry for not being clear.

Now, what I suggest is: In **"Mark"** dialog, it should have a **"color piker"** button/feild that return any color we what, and it will be used for highlighting the text (in-stand of the red color).

**Comment by Ekopalypse** [comment]
This could be solved if mark feature would use [SCI_SETINDICATORVALUE.](https://www.scintilla.org/ScintillaDoc.html#SCI_INDICSETFLAGS).
In theory, a color box button next to **Mark all** button could be implemented to allow selecting the color.
Unfortunately, my c++ isn't good enough to implement it.

**Comment by kirsan31** [comment]
Very needed feature, was surprised that this is not possible :(
